daemon.go

```diff
@@ -210,6 +221,54 @@ func (d *daemon) runCidCheck(ctx context.Context, cidKey cid.Cid, ipniURL string
 		wg.Add(1)
 		go func(provider peer.AddrInfo, src string) {
 			defer wg.Done()
+			// Get http retrieval out of the way if this is such
+			// provider.
+			httpInfo, libp2pInfo := network.SplitHTTPAddrs(provider)
+			if len(httpInfo.Addrs) > 0 && httpRetrieval {
+
+				provOutput := providerOutput{
+					ID: provider.ID.String(),
+					DataAvailableOverBitswap: BitswapCheckOutput{
+						Enabled: false,
+					},
+					DataAvailableOverHTTP: HTTPCheckOutput{
+						Enabled: true,
+					},
+					Source: src,
+				}
+				for _, ma := range httpInfo.Addrs {
+					provOutput.Addrs = append(provOutput.Addrs, ma.String())
+				}
+
+				testHost, err := d.createTestHost()
+				if err != nil {
+					log.Printf("Error creating test host: %v\n", err)
+					return
+				}
+				defer testHost.Close()
+				httpCheck := checkHTTPRetrieval(ctx, testHost, cidKey, httpInfo)
+				provOutput.DataAvailableOverHTTP = httpCheck
+				if !httpCheck.Connected {
+					provOutput.ConnectionError = httpCheck.Error
+				}
+				for _, ma := range httpCheck.Endpoints {
+					provOutput.ConnectionMaddrs = append(provOutput.ConnectionMaddrs, ma.String())
+				}
+
+				mu.Lock()
+				out = append(out, provOutput)
```

Comment:
With this approach you end up creating a test host for both HTTP and Bitswap, and produce two separate entries in the returned response for the same peer.

Is there any reason to not reuse the same test host and also reuse the same `provOutput` struct?

Reply:
Fair enough, I can update that